kmip/services/server/engine.py

```diff
@@ -1556,3 +1561,61 @@ def _process_discover_versions(self, payload):
         )
 
         return response_payload
+
+    @_kmip_version_supported('1.2')
+    def _process_mac(self, payload):
+        self._logger.info("Processing operation: MAC")
+
+        unique_identifier = self._id_placeholder
+        if payload.unique_identifier:
+            unique_identifier = payload.unique_identifier.value
+
+        object_type = self._get_object_type(unique_identifier)
+
+        managed_object = self._data_session.query(object_type).filter(
+            object_type.unique_identifier == unique_identifier
+        ).one()
```

Comment:
In this case I think it's appropriate to pass `GET` as the operation to `_get_object_with_access_controls` instead of `MAC`. That way we ensure that only a user with read access to the secret can use it to compute a MAC value.